src/webkit/wkPage.ts

```diff
@@ -677,6 +677,14 @@ export class WKPage implements PageDelegate {
       return;
     const response = request.createResponse(event.response);
     this._page._frameManager.requestReceivedResponse(response);
+
+    if (response.status() === 204) {
```

Comment:
isn't this the case only for navigation requests?

Reply:
Yes, it is only returned for main requests